nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -1233,6 +1233,13 @@ def check_branching_transition(
         dom_op_quantizers = set()
         for op_node_key in dom_op_node_keys:
             op_node = quant_prop_graph.nodes[op_node_key]
+
+            # Check all branches have a quantizer on it before the merge
+            if op_node["op_meta"].target_input_ports is not None:
+                all_branches_are_quantized = quant_prop_graph.all_outputs_are_quantized(branching_node_key)
+                if not all_branches_are_quantized:
+                    return TransitionStatus.SHOULD_NOT_TRANSITION
```

Comment:
Are you sure that we should in `check_brancking_transition` deny the merging? Also, there exists more appropriate status like `TransitionStatus.SHOULD_MERGE`.
Major.

Reply:
My changes are based on https://github.com/openvinotoolkit/nncf/pull/1071, which is quite similar to my target case: [This ](https://github.com/openvinotoolkit/nncf/pull/1071/files#diff-39f33773245f182909c8b9227aa638aac9e4bb39837264477d9a99cf92c11b31R1219-R1222)check was added to prevent outputs quantization (please see comment https://github.com/openvinotoolkit/nncf/pull/1071#issuecomment-1403654945). As you can see, `TransitionStatus.SHOULD_NOT_TRANSITION` was used here. Status `TransitionStatus.SHOULD_MERGE` will merge quantizer and will put them on the post hook of the branching node. In the target case of this pr that means that third branch of the `ScaledDotProductAttention` is being wrongly quantized